tests/orm/nodes/data/test_kpoints.py

```diff
@@ -81,3 +84,78 @@ def test_get_kpoints(self):
         kpt2 = load_node(kpt.pk)
         assert np.abs(kpt2.get_kpoints() - np.array(kpoints)).sum() == 0.0
         assert np.abs(kpt2.get_kpoints(cartesian=True) - np.array(cartesian_kpoints)).sum() == 0.0
+
+
+@skip_atomistic
+class TestKpointsAtomisticStructureData:
+    """Test for the `Kpointsdata` class using the new atomistic StructureData."""
+
+    @pytest.fixture(autouse=True)
+    def init_profile(self):
+        """Initialize the profile."""
+
+        from aiida_atomistic import StructureData, StructureDataMutable
+
+        alat = 5.430  # angstrom
+        cell = [
+            [
+                0.5 * alat,
+                0.5 * alat,
+                0.0,
+            ],
+            [
+                0.0,
+                0.5 * alat,
+                0.5 * alat,
+            ],
+            [0.5 * alat, 0.0, 0.5 * alat],
+        ]
+        self.alat = alat
+        mutable = StructureDataMutable()
+        mutable.set_cell(cell)
+        mutable.add_atom(positions=(0.000 * alat, 0.000 * alat, 0.000 * alat), symbols='Si')
+        mutable.add_atom(positions=(0.250 * alat, 0.250 * alat, 0.250 * alat), symbols='Si')
+        self.structure = StructureData.from_mutable(mutable)
```

Comment:
This can be moved as a `generate_atomistic_structure_data()` when called it return this example structure. 
I think there is also quite a bit duplicated code between `TestKpoints` and `TestKpointsAtomisticStructureData`. My suggestion is to combine these to above and use https://docs.pytest.org/en/stable/how-to/parametrize.html that same test logic works for both of them.